File: 3185/main.go
package main

import "fmt"

func main() {
	fmt.Println(countCompleteDayPairs([]int{12,12,23,43,24}))
}

func countCompleteDayPairs(hours []int) int64 {
	var ans int64
	m := make([]int, 24)
	for _, h := range hours {
		ans += int64(m[(24 - h % 24) % 24])
		m[h % 24]++
	}
	return ans
}

File: 3185/main.rs
struct Solution {}

impl Solution {
    pub fn count_complete_day_pairs(hours: Vec<i32>) -> i64 {
        let mut ans: i64 = 0;
        let mut m = vec![0;24];
        for h in hours.iter() {
            ans += m[((24-h%24)%24) as usize] as i64;
            m[(h%24) as usize] += 1;
        }
        ans
    }
}

fn main() {
    println!("{}", Solution::count_complete_day_pairs(vec![12,12,48,22,24]));
}